src/testdir/test_vim9_class.vim

```diff
@@ -518,6 +518,101 @@ def Test_class_default_new()
   v9.CheckScriptFailure(lines, 'E119:')
 enddef
 
+
+def Test_class_new_with_object_member()
+  var lines =<< trim END
+      vim9script
+
+      class C
+        this.str: string
+        this.num: number
+        def new(this.str, this.num)
+        enddef
+      endclass
+
+      def Check()
+        try
+          var c = C.new('cats', 2)
+          assert_equal('cats', c.str)
+          assert_equal(2, c.num)
+        catch
+          echo $'Exception was caught: {v:exception}'
+        endtry
+      enddef
+
+      Check()
+  END
+  v9.CheckScriptSuccess(lines)
+
+  lines =<< trim END
+      vim9script
+
+      class C
+        this.str: string
+        this.num: number
+        def newVals(this.str, this.num)
+        enddef
```

Comment:
The difference between this test and the previous test is that instead of using the new() constructor function, this test uses the  newVals() member function?  Is that correct?  Can these tests be combined into a single test?

Reply:
combined.